### PR TITLE
stage2 Sema: implement @intToPtr

### PR DESCRIPTION
**Pull request by g-w1**
I am not sure how to test this and the `int_u64` checking feels a little weird, can anyone think of any better way to represent a Value pointer?

### CODE REVIEW

File: src/type.zig [hunk 1]
```diff
@@ -1530,15 +1530,15 @@ pub const Type = extern union {
             .const_slice,
             .mut_slice,
             => {
-                if (self.elemType().hasCodeGenBits()) return @divExact(target.cpu.arch.ptrBitWidth(), 8) * 2;
+                if (!self.elemType().hasCodeGenBits()) return @divExact(target.cpu.arch.ptrBitWidth(), 8) * 2;
```
Comment (LemonBoy) [inline src/type.zig:1533]: This looks wrong, if the element type is non-zero sized you have both a pointer and a length field, notice the `* 2` at the end.

Comment (g-w1) [reply src/type.zig:1533]: Ah, for the other ones I think it was a typo, but for this one it was not. Thanks

File: src/Sema.zig [hunk 2]
```diff
@@ -5785,6 +5785,14 @@ fn zirIntToPtr(sema: *Sema, block: *Scope.Block, inst: Zir.Inst.Index) InnerErro
         });
     }
     try sema.requireRuntimeBlock(block, src);
+    if (block.wantSafety()) {
+        const zero = try sema.mod.constInst(sema.arena, src, .{
+            .ty = Type.initTag(.u64),
+            .val = Value.initTag(.zero),
+        });
+        const is_non_zero = try block.addBinOp(src, Type.initTag(.bool), .cmp_neq, operand_coerced, zero);
```
Comment (LemonBoy) [inline src/Sema.zig:5793]: This is wrong as some pointer types do accept zero as value. Check out `ir_analyze_int_to_ptr`, the pointer alignment must be validated too.

Comment (Vexu) [reply src/Sema.zig:5799]: Right, thanks fixed.

**Pull request merged**